posts/building-a-realtime-app-with-phoenix.md

```diff
@@ -0,0 +1,604 @@
+
+```javascript
+channel.on("response:updated", (response) => {
+  // do stuff with response.challenge
+})
+```
+
+And that's it! Now that we've build out our challenge channel and starting broadcasting messages, we're ready to move on to Part II: tracking user presence in the channel. 
+
+## Part II: Tracking User State with Phoenix Presence
+
+We already have the ability for users to broadcast messages to all of the clients subscribing to a channel. But we want to give our users awareness of who these clients are. We want to list out the people currently in a given "challenge room" *and* we want to indicate which user is currently typing into the shared text editor. 
+
+We'll use Phoenix Presence to store and expose this user-state-related info. 
+
+## What is Phoenix Presence?
+The [Phoenix Presence module](https://hexdocs.pm/phoenix/Phoenix.Presence.html) allows us to:
+
+* Store and expose topic-specific information to all of a channel's subscribing clients, across all of our application's distributed nodes. 
+* Store that information in a resilient and de-centralized manner.
+* Broadcast presence-related events and handle them on the front-end with ease. 
+
+## How Does Phoenix Presence Work?
+
+Presence uses a [Conflict-Free Replicated Data Type (CRDT)](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type) to store and broadcast topic-specific information to all channels with a given topic. Unlike a centralized data store like Redis, Phoenix Presence has no single point of failure and no single source of truth. This allows the Presence data store to be self-healing and replicable across all of the nodes of your application cluster. 
+
+
+## Implementing Phoenix Presence
+
+We'll use Phoenix Presence to support two real-time, user presence related features:
+
+* Real-time list of users present in a given "challenge room"
+* Indication of who is typing in the shared text editor
+
+In order to implement these two features with Phoenix Presence, we need to do three things:
+
+* Incorporate the Phoenix Presence module into our app
+* Use Phoenix Presence to store the users in a given "challenge room"
+* Use Phoenix Presence to track who is currently typing in the shared text editor. 
+
+Let's do it!
+
+### Setting Up Phoenix Presence
+
+Incorporating Phoenix Presence in our app is easy. We need to:
+
+* Define a module that uses Presence
+* Include that module in our supervisor tree
+
+We'll define our module `ChallengePresence` in `lib/phoenix_pair`. We'll tell our module to `use` the `Phoenix.Presence` module and we'll provide our module with access to our OTP app and PubSub server.
+
+```elixir
+defmodule PhoenixPair.ChallengePresence do
+  use Phoenix.Presence, otp_app: :phoenix_pair,
+                        pubsub_server: PhoenixPair.PubSub
+
+  alias PhoenixPair.{ChallengePresence}
```

Comment:
Same here should be using a singular alias and not the multi variant and is there a reason there is an alias to the current module?